utils/protocols.py

```diff
@@ -1,13 +1,35 @@
-ONTOLOGY_10x = ["EFO:0009310", "EFO:0009899"]
+from api.ontology import OntologyAPI
 
+ONTOLOGY_10x_PARENT = "EFO:0008995"
+ONTOLOGY_10xV2_PARENT = "EFO:0009310"
+ONTOLOGY_10xV3_PARENT = "EFO:0009898"
 
-def is_10x(library_preparation_protocol):
+
+def is_10x(ontology_api: OntologyAPI, library_preparation_protocol : dict):
+    library_const_approach = get_library_const_approach(library_preparation_protocol)
+    if library_const_approach:
+        if ONTOLOGY_10x_PARENT == library_const_approach:
+            return True
+        if ontology_api.is_child_of(ONTOLOGY_10x_PARENT, library_const_approach):
+            return True
+    return False
+
```

Comment:
if library construction approach is empty, I think it's better to throw an error. It's a required field in the schema it should never be empty. The user must provide this info. If this is not provided in the metadata, something must have gone wrong in the validation and that should be flagged to the user.

Reply:
Good point, I'll make this change.